course5/homework06/string.cc

```diff
@@ -0,0 +1,150 @@
+#include "string.h"
+#include<memory>
+#include<algorithm>
+#include<cstring>
+
+String::String()
```

Comment:
This constructor leaves ```size``` uninitialised.
You should either provide the default value in the class' definition (better option), or set the default value here (less nice option)